connect-inject/endpoints_controller.go

```diff
@@ -247,6 +180,103 @@ func (r *EndpointsController) SetupWithManager(mgr ctrl.Manager) error {
 		).Complete(r)
 }
 
+// registerServicesAndHealthCheck create Consul registrations for the service and proxy and register them with Consul.
```

Comment:
Minor nit :
```suggestion
// registerServicesAndHealthCheck will create Consul registrations for the service and proxy and register them with Consul.
```

This is completely a nonblocking comment but I was wondering about the name of the function in that we're not "registering" the health check (actually we are in the case it doesn't exist) but rather its an Upsert process, so maybe we could call it `registerServicesAndUpsertHealthCheck()` or `registerServicesAndUpdateHealthCheck` or something?
I think it is fine if we dont make a change but wondered if you had given any minor thought to it?

Reply:
I think it's OK to refer to both as `register`. Looking at Consul API for [health checks](https://www.consul.io/api-docs/agent/check#register-check), it seems like upserting a health check is the same as registering a health check, so I think this name is fine.

There is a distinction in that we don't always register health check and just update the health status, but I don't think this needs to be reflected in the name of the function necessarily.

Reply:
Sweet, thanks! Yeah I was on the fence about even mentioning it but thought I'd ask if you'd thought about it. Sounds good!

Reply:
I'm glad you brought it up!